docs/src/dplyr.md

```diff
@@ -0,0 +1,293 @@
+
+## Selecting Columns Using `@select`
+
+Select a set of columns: the `:name` and the `:sleep_total` columns. 
+
+```@example 1
+sleepData = @select msleep :name :sleep_total
+```
+
+To select all the columns *except* a specific column, use the `Not` function for inverse selection. We preface the `Not` with `$` because it does not reference a column directly as a `Symbol`.
+
+```@example 1
+@select msleep $(Not(:name))
+```
+
+To select a range of columns by name, use the `Between` operator:
+
+```@example 1
+@select msleep $(Between(:name, :order))
+```
+
+To select all columns that start with the character string `"sl"` use regular expressions. 
+
+```@example 1
+@select msleep $(r"^sl")
+```
+
+Regular expressions are powerful, but can be difficult for new users to understand. Here are some quick tips.  
+
+1. `r"^abc"` = Starts with `"abc"`
+2. `r"abc$"` = Ends with `"abc"`
+3. `r"abc"` = Contains `"abc"` anywhere. 
+
+## Selecting Rows Using `@subset` and `@rsubset`
+
+Filter the rows for mammals that sleep a total of more than 16 hours. 
+
+```@example 1
+@subset msleep :sleep_total .>= 16
+```
+
+In the above expression, the `.>=` means we "broadcast" the `>=` comparison across the whole column. We can use a simpler syntax, `@rsubset` which automatically broadcasts all operations.
+
+```@example 1
+@rsubset msleep :sleep_total > 16
+```
+
+Subset the rows for mammals that sleep a total of more than 16 hours *and* have a body weight of greater than 1 kilogram. For this we put multiple operations on separate lines in a single block. 
+
+```@example 1
+@rsubset msleep begin 
+    :sleep_total >= 16 
+    :bodywt >= 1
+end
+```
+
+Filter the rows for mammals in the Perissodactyla and Primates taxonomic order
+
+```@example 1
+@rsubset msleep :order in ["Perissodactyla", "Primates"]
+```
+
+You can use the boolean operators (e.g. >, <, >=, <=, !=, in) to create the logical tests. 
+
+# Chain.jl
+
+Before we go any further, let's introduce the `@chain` macro from [Chain.jl](https://github.com/jkrumbiegel/Chain.jl), which is re-exported by DataFramesMeta.jl. `@chain` allows for you to pipe the output from one operation into the input of another operation.  The idea of piping is to read the functions from left to right. The syntax and design of `@chain` is very similar to `%>%` which users of dplyr are familiar with.
+
+To show it's usefulness, let's use `@select` and `@rsubset` one after the other. 
+
+```
+msleep_1 = @select msleep :name :sleep_total
+msleep_2 = @rsubset msleep_1 :sleep_total > 16
+```
+
+Now in this case, we will pipe the msleep data frame to the function that will select two columns (name and sleep\_total) and then pipe the new data frame to the `@rsubset` opertaion. This method involves awkwardly creating and naming temporary data frames. This can be avoided with `@chain`.
```

Comment:
```suggestion
Now in this case, we will pipe the msleep data frame to the function that will select two columns (`:name` and `:sleep_total`) and then pipe the new data frame to the `@rsubset` operation. This method involves awkwardly creating and naming temporary data frames. This can be avoided with `@chain`.
```